File: src/main/java/com/basil/shoppingcart/service/AdminOrderService.java
package com.basil.shoppingcart.service;

import java.util.List;

import com.basil.shoppingcart.dto.request.UpdateOrderStatusRequest;
import com.basil.shoppingcart.dto.response.OrderResponse;

public interface AdminOrderService {

    List<OrderResponse> getAllOrders();

    OrderResponse getOrderById(Long orderId);

    OrderResponse updateOrderStatus(Long orderId,UpdateOrderStatusRequest request);
}

File: shopping-cart-frontend/src/services/adminOrderService.js
import apiClient from "./apiClient";

const adminOrderService = {
    getAllOrders: async () => {
        const response = await apiClient.get("/admin/orders");
        return response.data;
    },

    getOrderById: async (orderId) => {
        const response = await apiClient.get(`/admin/orders/${orderId}`);
        return response.data;
    },

    updateOrderStatus: async (orderId, status) => {
        const response = await apiClient.put(
            `/admin/orders/${orderId}/status`,
            { status }
        );
        return response.data;
    }
};

export default adminOrderService;